### PR TITLE
fixes #13638 - Enable rubocop 0.37.1 cops

### PR DESCRIPTION
**Pull request by brandonweeks**
Depends on #373.

### CODE REVIEW

File: lib/proxy/log.rb [hunk 1]
```diff
@@ -27,13 +27,13 @@ def self.default_logger(log_file)
 
     def self.logger
       log_file = ::Proxy::SETTINGS.log_file
-      if log_file.upcase == 'STDOUT'
+      if log_file.casecmp('STDOUT')
```
Comment (domcleal) [inline lib/proxy/log.rb]: This returns an int, it needs comparing to zero I think. You could add `.zero?`

File: lib/proxy/log.rb [hunk 2]
```diff
@@ -27,13 +27,13 @@ def self.default_logger(log_file)
...
         if SETTINGS.daemon
           puts "Settings log_file=STDOUT and daemon=true are incompatible, exiting..."
           exit 1
         end
         logger = ::Logger.new(STDOUT)
-      elsif log_file.upcase == 'SYSLOG'
+      elsif log_file.casecmp('SYSLOG')
```
Comment (domcleal) [inline lib/proxy/log.rb]: ditto, `.zero?`

File: tasks/pkg.rake [hunk 3]
```diff
@@ -6,7 +6,7 @@ namespace :pkg do
     File.exist?('pkg') || FileUtils.mkdir('pkg')
     ref = ENV['ref'] || 'HEAD'
     version = `git show #{ref}:VERSION`.chomp.chomp('-develop')
-    raise "can't find VERSION from #{ref}" if version.length == 0
+    raise "can't find VERSION from #{ref}" if !version.lengt.empty?
```
Comment (domcleal) [inline tasks/pkg.rake]: typo, `!version.empty?`

File: test/dhcp_isc/server_isc_test.rb [hunk 4]
```diff
@@ -64,15 +64,15 @@ def test_sparc_host_quirks
...
-      %q{option SUNW.root-server-hostname \"itgsyddev807.macbank\";},
-      %q{option SUNW.root-server-ip-address 192.168.122.24;},
-      %q{option SUNW.sysid-config-file-server \"192.168.122.24:/Solaris/jumpstart/sysidcfg/sysidcfg_primary\";},
-      %q{vendor-option-space SUNW;}
+      'option SUNW.JumpStart-server \"192.168.122.24:/Solaris/jumpstart\";',
+      'option SUNW.install-path \"/Solaris/install/Solaris_5.10_sparc_hw0811\";',
+      'option SUNW.install-server-hostname \"itgsyddev807.macbank\";',
+      'option SUNW.install-server-ip-address 192.168.122.24;',
+      'option SUNW.root-path-name \"/Solaris/install/Solaris_5.10_sparc_hw0811/Solaris_10/Tools/Boot\";',
+      'option SUNW.root-server-hostname \"itgsyddev807.macbank\";',
```
Comment (domcleal) [inline test/dhcp_isc/server_isc_test.rb]: I think the meaning of the escapes may change here, as `'\"'` is the value `\"` while `%q{\"}` is `"`.

**Comment by domcleal** [comment]
Sorry for the duplicate comments, I incorrectly commented on the commit rather than the PR, so my comments would get lost on GitHub.

**Comment by dmitri-d** [comment]
I'm still thinking about this, but I'm not sure how much I like enforcing coding style to such a degree...

**Comment by domcleal** [comment]
The cops I'm particularly supportive of are the ones with a performance impact - those under Performance/, but also some stylistic ones that have a small impact.  Some of the whitespace type ones can end up in excessive bikeshedding, but otherwise I think the consistency it brings to the project - both internally, and with the wider community - is worth it.

**Comment by dLobatog** [comment]
I'm fine with nearly of all of it. The ABC complexity, class size and other cops are a bit hard to enforce but overall I like how the code looks the same no matter where you look with rubocop.

**Comment by brandonweeks** [comment]
[test]

File: test/dhcp_isc/server_isc_test.rb (hunk 4 above)
Comment (domcleal) [inline test/dhcp_isc/server_isc_test.rb]: Unfixed?

**Comment by domcleal** [comment]
Would you mind rebasing this?  There's also an inline comment about escaping that I'm concerned about.

File: test/dhcp_isc/server_isc_test.rb (hunk 4 above)
Comment (dmitri-d) [inline test/dhcp_isc/server_isc_test.rb]: This is an issue indeed.

**Comment by brandonweeks** [comment]
Done. :)

**Comment by domcleal** [comment]
Merged as 5f5c216, thanks @brandonweeks.

**Pull request closed without merging**